File: code/backend/foodometer-order-service/src/main/java/com/springmicro/foodometer/document/FoodOrder.java
package com.springmicro.foodometer.document;

import com.springmicro.foodometer.constants.FoodOrderStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Document("FoodOrder")
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class FoodOrder {
    private String id;
    private List<FoodItemQuantity> foodItems;
    private String customerId;
    private String customerAddressId;
    private LocalDateTime orderTimestamp;
    private Double orderAmount;
    private FoodOrderStatus orderStatus;
    private Double discount;
    private Double payableAmount;
}

File: code/frontend/food-o-gui/src/app/_model/food-order.ts
import { FoodItemQuantity } from "./food-item-quantity";

export class FoodOrder {
    id: string;
    foodItems: FoodItemQuantity[];
    orderTimestamp: Date;
    orderAmount: number;
    orderStatus: string;
    discount: number;
    payableAmount: number;
    customerId: string;
    customerAddressId: string;

    constructor(
        customerId?: string,
        customerAddressId?: string,
        foodItems?: FoodItemQuantity[],
        orderTimestamp?: Date,
        orderAmount?: number,
        orderStatus?: string,
        discount?: number,
        payableAmount?: number,
        id?: string) {
        this.id = id;
        this.foodItems = foodItems;
        this.orderTimestamp = orderTimestamp;
        this.orderAmount = orderAmount;
        this.orderStatus = orderStatus;
        this.discount = discount;
        this.payableAmount = payableAmount;
        this.customerId = customerId;
        this.customerAddressId = customerAddressId;
    }
}
